test/infamy/topology.py

```diff
@@ -11,24 +13,36 @@ def _qstrip(text):
 
     return text
 
+def satisfies(provides, requires):
+    provides_set = set(provides.split() if isinstance(provides, str) and provides else provides)
+    requires_set = set(requires.split() if isinstance(requires, str) and requires else requires)
```

Comment:
```suggestion
    provides_set = set(provides.split())
    requires_set = set(requires.split())
```
Why is this not enough here? Can `provides` or `requires` ever _not_ be strings?

Reply:
It is actually enough. Thnx.
The whole complexity came out of the "requires" part sent from get_ctrl function. I was unnecessarily sending a list [ ] type instead of string, so had to support that afterwards.